src/app/CommandHandler.h

```diff
@@ -411,31 +406,23 @@ class CommandHandler : public Messaging::ExchangeDelegate
     Access::SubjectDescriptor GetSubjectDescriptor() const
     {
         VerifyOrDie(!mGoneAsync);
-        return mExchangeCtx->GetSessionHandle()->GetSubjectDescriptor();
+        return mDispatcher.GetSubjectDescriptor();
     }
 
 private:
     friend class TestCommandInteraction;
     friend class CommandHandler::Handle;
-
-    CHIP_ERROR OnMessageReceived(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
-                                 System::PacketBufferHandle && payload) override;
-
-    void OnResponseTimeout(Messaging::ExchangeContext * ec) override
-    {
-        //
-        // We're not expecting responses to any messages we send out on this EC.
-        //
-        VerifyOrDie(false);
-    }
+    // TODO would be much better if we provided a Closure instead of using friend here.
```

Comment:
I am hoping the ugliness of adding this friend class is outweighed by Dispatching being more encapsulated and that this effort can be left to a future clean up